### PR TITLE
backend/enh : automates payouts for metro incentive

### PR DESCRIPTION
**Pull request by shreyaxh**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/dhall-configs/dev/rider-app.dhall [hunk 1]
```diff
@@ -199,6 +200,7 @@ let jobInfoMapx =
         , mapValue = False
         }
       , { mapKey = RiderJobType.OtherJobTypes, mapValue = False }
+      , { mapKey = AllocatorJobType.MetroIncentivePayout, mapValue = True }
```
Comment (prashant601) [inline Backend/dhall-configs/dev/rider-app.dhall:203]: please add this in triggerScheduler api to trigger this job

Comment (shreyaxh) [reply Backend/dhall-configs/dev/rider-app.dhall:203]: We will run a query to add jobData to Db directly

File: Backend/app/rider-platform/rider-app/Main/src-read-only/Domain/Types/PayoutConfig.hs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+data PayoutConfig = PayoutConfig
+  { batchLimit :: Kernel.Prelude.Int,
+    isPayoutEnabled :: Kernel.Prelude.Bool,
+    maxRetryCount :: Kernel.Prelude.Int,
+    merchantId :: Kernel.Types.Id.Id Domain.Types.Merchant.Merchant,
+    merchantOperatingCityId :: Kernel.Types.Id.Id Domain.Types.MerchantOperatingCity.MerchantOperatingCity,
+    orderType :: Kernel.Prelude.Text,
+    payoutEntity :: Domain.Types.PayoutConfig.PayoutEntity,
+    payoutRegistrationCgst :: Kernel.Types.Common.HighPrecMoney,
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src-read-only/Domain/Types/PayoutConfig.hs:23]: Remove `payoutRegistrationCgst`, `payoutRegistrationFee`, `payoutRegistrationSgst`, `thresholdPayoutAmountPerOrder1

**Review by NikithShetty — CHANGES_REQUESTED** [review]

File: Backend/dev/migrations-read-only/rider-app/frfs_ticket_booking.sql [hunk 3]
```diff
@@ -72,4 +72,16 @@ ALTER TABLE atlas_app.frfs_ticket_booking ALTER COLUMN merchant_id SET NOT NULL;
...
 ALTER TABLE atlas_app.frfs_ticket_booking ADD COLUMN event_discount_amount double precision ;
-ALTER TABLE atlas_app.frfs_ticket_booking ADD COLUMN discounted_tickets integer ;
\ No newline at end of file
+ALTER TABLE atlas_app.frfs_ticket_booking ADD COLUMN discounted_tickets integer ;
+
+
+------- SQL updates -------
+
+ALTER TABLE atlas_app.frfs_ticket_booking ADD COLUMN payer_vpa text ;
+ALTER TABLE atlas_app.frfs_ticket_booking ADD COLUMN cashback_settled boolean ;
```
Comment (NikithShetty) [inline Backend/dev/migrations-read-only/rider-app/frfs_ticket_booking.sql:81]: we can use `cashback_status` instead

File: Backend/app/rider-platform/rider-app/Main/src/Tools/Payment.hs [hunk 4]
```diff
@@ -188,4 +189,9 @@ runWithServiceConfig3 func getCfg merchantId merchantOperatingCityId req1 req2 r
 data PaymentServiceType = Normal | FRFSBooking
   deriving (Generic, FromJSON, ToJSON, Show, ToSchema, ToParamSchema)
 
+data ServiceNames = MetroCashback | Other
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src/Tools/Payment.hs:192]: `ServiceNames` not used, can be removed

File: Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/FRFSTicketBookingExtra.hs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      updatePayerVpaByBookingId bookingId vpa
+  pure ()
+insertPayerVpaIfNotPresent _ _ = pure ()
+
+updatePayerVpaByBookingId :: (EsqDBFlow m r, MonadFlow m, CacheFlow m r) => Id FRFSTicketBooking -> Text -> m ()
+updatePayerVpaByBookingId id payerVpa = do
+  now <- getCurrentTime
+  updateOneWithKV [Se.Set Beam.payerVpa (Just payerVpa), Se.Set Beam.updatedAt now] [Se.Is Beam.id $ Se.Eq (id.getId)]
+
+findAllByCashbackSettled :: (EsqDBFlow m r, MonadFlow m, CacheFlow m r) => Maybe Int -> Maybe Int -> Maybe Bool -> Maybe CashbackStatus -> m [FRFSTicketBooking]
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/FRFSTicketBookingExtra.hs:34]: rename to `findAllByCashbackStatus`

**Review by NikithShetty — CHANGES_REQUESTED** [review]

**Pull request merged**